cmake/cmaize/package_managers/pip/pip.cmake

```diff
@@ -0,0 +1,165 @@
+    # :return: An empty string if ``package_specs`` is not already installed,
+    #    otherwise a target representing the package.
+    # :rtype: string or InstalledTarget
+    #]]
+    cpp_member(find_installed PIPPackageManager desc PackageSpecification args)
+    function("${find_installed}" self _fi_result _fi_package_specs)
+
+        _pip_find_installed(
+            "${self}" "${_fi_result}" "${_fi_package_specs}"
+        )
+
+    endfunction()
+
+    #[[[
+    # Retrieves an already installed package.
+    #
+    # This method is a convenience function for calling find_installed and
+    # asserting that the result is not an empty string.
+    #
+    # :param self: The pip package manager to check for the pacakge.
+    # :type self: PIPPackageManager
+    # :param result: A variable to hold the resulting target
+    # :type result: InstalledTarget*
+    # :param package_specs: The details of the package we are looking for.
+    # :type package_specs: PackageSpecification.
+    # :return: A target detailing the found package.
+    # :rtype: InstalledTarget
+    #
+    # :raises PACKAGE_NOT_FOUND: When ``package_specs`` is not already
+    #    installed.
+    #]]
+    cpp_member(get_package PIPPackageManager str PackageSpecification args)
+    function("${get_package}" self _gp_result _gp_package_specs)
+
+        _pip_get_package(
+            "${self}" "${_gp_result}" "${_gp_package_specs}" ${ARGN}
+        )
+
+    endfunction()
+
+    #[[[
+    # Installs the specified package.
+    #
+    # .. todo:
+    #
+    #    This should take a PackageSpecification and install the package during
+    #    the build phase.
+    #
+    # This method is used to have the package manager install a package. At the
+    # moment, the installation happens at configuration time.
+    #
+    # :param self: The package manager the package will be installed into.
+    # :type self: PIPPackageManager
+    # :param package_specs: The specs for the package we need to install.
+    # :type package_specs: PackageSpecification
+    #]]
+    cpp_member(install_package PIPPackageManager PackageSpecification args)
+    function("${install_package}" self _ip_package_specs)
+
+        _pip_install_package("${self}" "${_ip_package_specs}" ${ARGN})
+
+    endfunction()
+
+
+cpp_end_class()
+
+#[[[
+# Enables and registers a pip package manager with the global environment.
+#
+# The pip package manager is only registered if "pip" has not already been
+# added to the CMAIZE_SUPPORTED_PACKAGE_MANAGERS global variable.
+#
+# .. note::
+#
+#    Enabling the pip package manager adds Python as a dependency of CMaize.
+#]]
+function(enable_pip_package_manager)
+
+    cpp_get_global(_eppm_pms CMAIZE_SUPPORTED_PACKAGE_MANAGERS)
+    if("pip" IN_LIST _eppm_pms) # If pip is in the list, it's already enabled
+        return()
+    endif()
+
+    set(_eppm_pms "${_eppm_pms}" "pip")
+    cpp_set_global(CMAIZE_SUPPORTED_PACKAGE_MANAGERS "${_eppm_pms}")
```

Comment:
See my comment on the same code pattern in `docs/src/developer/new_package_manager.rst`.